app/models/course/announcement.rb

```diff
@@ -19,4 +22,14 @@ def set_defaults
   def valid_in_future?
     valid_from > DateTime.now
   end
+
+  # Get the number of unread announcements for current user in current course
+  #
+  # @param current_user [User] The login user.
+  # @param course [Course] The Course which user is visiting
+  # @return [Integer] The number of unread announcements.
+  def self.get_unread_number(current_user, course)
+    return unless current_user && course
+    unread_by(current_user).where(course_id: course.id).count
```

Comment:
`unread_by` is a scope. You can chain scopes with other relations. `course.announcements.unread_by(current_user)` is valid. There is not need for this helper, IMO.